aiperf/post_processors/timeslice_metric_results_processor.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+from collections import defaultdict
+from typing import Any
+
+from pygments.lexer import default
+
+from aiperf.common.config import UserConfig
+from aiperf.common.constants import NANOS_PER_MILLIS
+from aiperf.common.decorators import implements_protocol
+from aiperf.common.enums import ResultsProcessorType
+from aiperf.common.exceptions import NoMetricValue, PostProcessorDisabled
+from aiperf.common.factories import ResultsProcessorFactory
+from aiperf.common.messages.inference_messages import MetricRecordsData
+from aiperf.common.models import MetricResult
+from aiperf.common.protocols import ResultsProcessorProtocol
+from aiperf.common.types import MetricTagT, TimeSliceT
+from aiperf.metrics.base_metric import BaseMetric
+from aiperf.metrics.metric_dicts import MetricResultsDict
+from aiperf.metrics.metric_registry import MetricRegistry
+from aiperf.post_processors.metric_results_processor import MetricResultsProcessor
+
+
+@implements_protocol(ResultsProcessorProtocol)
+@ResultsProcessorFactory.register(ResultsProcessorType.TIMESLICE)
+class TimesliceMetricResultsProcessor(MetricResultsProcessor):
+    """Processor for metric results in timeslice mode.
+
+    Groups metrics by time slices based on request timestamps and slice_duration.
+    """
+
+    def __init__(self, user_config: UserConfig, **kwargs: Any):
+        super().__init__(user_config=user_config, **kwargs)
+
+        if self.user_config.output.slice_duration is None:
+            raise PostProcessorDisabled(
+                "TimesliceMetricResultsProcessor requires slice_duration to be set"
+            )
+
+        # Set up aggregate metric object default initialization for each timeslice
+        self._timeslice_instances_maps: dict[
+            TimeSliceT, dict[MetricTagT, BaseMetric]
+        ] = defaultdict(
+            lambda: {
+                tag: MetricRegistry.get_class(tag)()
+                for tag in MetricRegistry.all_tags()
+            }
+        )
+
+        # Use instance variable with defaultdict for auto-vivification
+        self._timeslice_results: dict[TimeSliceT, MetricResultsDict] = defaultdict(
+            MetricResultsDict
+        )
+
+    async def get_instances_map(
+        self, record_data: MetricRecordsData = None
+    ) -> dict[MetricTagT, BaseMetric]:
+        """Get the appropriate instances map based on mode."""
+        """Get the results dict for the appropriate timeslice based on request timestamp."""
+        # Extract the request start time from the record metadata
+        request_start_ns = record_data.metadata.request_start_ns
+
+        # Calculate timeslice index
+        # slice_duration is in milliseconds, timestamps are in nanoseconds
+        slice_duration_ns = self.user_config.output.slice_duration * NANOS_PER_MILLIS
```

Comment:
this computation and lookup is done a lot of places. you should cache the 
```python
self.slice_duration_ns = self.user_config.output.slice_duration * NANOS_PER_MILLIS
```
in the `__init__()` and use `self.slice_duration_ns` everywhere.